drivers/source/CAN.cpp

```diff
@@ -46,6 +46,30 @@ CAN::CAN(PinName rd, PinName td, int hz) : _can(), _irq()
     can_irq_init(&_can, (&CAN::_irq_handler), (uint32_t)this);
 }
 
+CAN::CAN(const can_pinmap_t &pinmap) : _can(), _irq()
+{
+    // No lock needed in constructor
+
+    for (size_t i = 0; i < sizeof _irq / sizeof _irq[0]; i++) {
```

Comment:
This loop doesn't do anything? You're already initialising `_irq` with the `_irq()` in the initializer-list. Guess you copied it from the others. Scrub it.

(Don't even actually need the `_irq()`, as it's an array of `Callback`s with default constructors, just as you don't need a `_mutex()`).